prow/github/client_test.go

```diff
@@ -217,6 +217,91 @@ func TestIsMember(t *testing.T) {
 	}
 }
 
+func TestIsMergeable(t *testing.T) {
+	testCases := []struct {
+		name           string
+		mergeableState MergeableState
+		expectedResult bool
+		isErrExpected  bool
+	}{
+		{
+			name:           "should be true when MergeableState is clean",
+			mergeableState: MergeableStateClean,
+			expectedResult: true,
+			isErrExpected:  false,
+		},
+		{
+			name:           "should be true when MergeableState is unstable",
+			mergeableState: MergeableStateUnstable,
+			expectedResult: true,
+			isErrExpected:  false,
+		},
+		{
+			name:           "should be false when MergeableState is behind",
+			mergeableState: MergeableStateBehind,
+			expectedResult: false,
+			isErrExpected:  true,
+		},
+		{
+			name:           "should be false when MergeableState is blocked",
+			mergeableState: MergeableStateBlocked,
+			expectedResult: false,
+			isErrExpected:  true,
+		},
+		{
+			name:           "should be false when MergeableState is draft",
+			mergeableState: MergeableStateDraft,
+			expectedResult: false,
+			isErrExpected:  true,
+		},
+		{
+			name:           "should be false when MergeableState is unknown",
+			mergeableState: MergeableStateUnknown,
+			expectedResult: false,
+			isErrExpected:  true,
+		},
+	}
+	mergable := true
+	for _, tc := range testCases {
```

Comment:
This is not safe and should be flagged by the `scopelint` linter.

Instead prefer to switch to a `map[string]struct{}` for your test cases with the current `name` field being the key. Then the loop start becomes:

```suggestion
	for name := range testCases {
		tc := testCases[name]
```

Reply:
Ah, interesting - this structure was copy/pasted from another test in the same suite.  I don't think that the K8S linting regime includes scopelint, in this case.

I'll make the change. 👍